sdk/state/update.go

```diff
@@ -11,21 +11,22 @@ import (
 
 type Payment struct {
 	IterationNumber       int64
-	Amount                int64
+	Amount                Amount
 	CloseSignatures       []xdr.DecoratedSignature
 	DeclarationSignatures []xdr.DecoratedSignature
 	FromInitiator         bool
+	NewBalance            Amount
```

Comment:
I think I referred to this concept as 'new balance', but on seeing it in code this looks a little too much like a NewBalance function. I think it'd be fine to name this Balance.
```suggestion
	Balance               Amount
```
Wdyt?

Reply:
imo `payment.Balance` and `payment.Amount` sound very similar. What about `payment.ChannelBalance`?

Reply:
Maybe we shouldn't be storing payment, but instead storing an `Agreement` or `CloseState` or `CloseAgreement`.

A payment needs an iteration number, source, amount, and signatures.

A close agreement needs an iteration number, balance, and signatures.

We've got some overlap but these feel like different things.

The main reason we're storing a payment is to be able to encode channel state like balance and be clear about which payment/agreement/iteration they belong. We still get that if we store something like these as a different type with the specific fields we need.

Reply:
I agree CloseAgreement works better. Changed it here:
https://github.com/stellar/experimental-payment-channels/pull/68/commits/fac8cc922ab48591a87cfd903a80d5018c6c18ac